### ISSUE TITLE
nswag webapi2swagger for all controllers fails (command line)

### ISSUE THREAD

**Issue by CastleSoft** [NSwag/NSwag#79]
The command line for /controller:WebApplication1.Controllers.ValuesController works fine.
When I try to do all controllers (using empty) /controller I get the following:

```
nswag webapi2swagger /assembly:$(TargetPath) /output:$(ProjectDir)\TypeScript\$(SolutionName).json
nswag swagger2tsclient /input:$(ProjectDir)\TypeScript\$(SolutionName).json /output:$(ProjectDir)\TypeScript\$(SolutionName).ts
```

Exception:

```
The Web API controller full class name or empty to load all controllers from the assembly.
1>  Controller: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: Could not find item type of array type 'System.Web.Mvc.TempDataDictionary'.
1>  
1>  Server stack trace: 
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.LoadProperty(Type parentType, PropertyInfo property, JsonSchema4 parentSchema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GeneratePropertiesAndInheritance(Type type, JsonSchema4 schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.RootTypeJsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\RootTypeJsonSchemaGenerator.cs:line 45
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.RootTypeJsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\RootTypeJsonSchemaGenerator.cs:line 52
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.LoadProperty(Type parentType, PropertyInfo property, JsonSchema4 parentSchema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GeneratePropertiesAndInheritance(Type type, JsonSchema4 schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.RootTypeJsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\RootTypeJsonSchemaGenerator.cs:line 45
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.CreateAndAddSchema(SwaggerService service, Type type, IEnumerable`1 parentAttributes, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 499
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.LoadReturnType(SwaggerService service, SwaggerOperation operation, MethodInfo method, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 452
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.GenerateForController(SwaggerService service, Type controllerType, String excludedMethodName, SchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 98
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.GenerateForControllers(IEnumerable`1 controllerTypes, String excludedMethodName) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 72
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiAssemblyToSwaggerGenerator.WebApiAssemblyLoader.GenerateForControllers(IEnumerable`1 controllerClassNames, String settingsData) in C:\projects\nswag\src\NSwag.CodeGeneration\SwaggerGenerators\WebApi\WebApiAssemblyToSwaggerGenerator.cs:line 89
1>     at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
1>     at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)
1>  
1>  Exception rethrown at [0]: 
1>     at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
1>     at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiAssemblyToSwaggerGenerator.WebApiAssemblyLoader.GenerateForControllers(IEnumerable`1 controllerClassNames, String settingsData)
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiAssemblyToSwaggerGenerator.GenerateForControllers(IEnumerable`1 controllerClassNames) in C:\projects\nswag\src\NSwag.CodeGeneration\SwaggerGenerators\WebApi\WebApiAssemblyToSwaggerGenerator.cs:line 62
1>     at NSwag.Commands.WebApiToSwaggerCommand.<RunAsync>d__32.MoveNext() in C:\projects\nswag\src\NSwag\Commands\WebApiToSwaggerCommand.cs:line 97
1>  --- End of stack trace from previous location where exception was thrown ---
1>     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
1>     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
1>     at NSwag.Commands.WebApiToSwaggerCommand.<RunAsync>d__31.MoveNext() in C:\projects\nswag\src\NSwag\Commands\WebApiToSwaggerCommand.cs:line 84
1>  --- End of stack trace from previous location where exception was thrown ---
1>     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
1>     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
1>     at NConsole.CommandLineProcessor.<ProcessAsync>d__8.MoveNext()
1>     --- End of inner exception stack trace ---
1>     at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
1>     at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
1>     at System.Threading.Tasks.Task`1.get_Result()
1>     at NSwag.Program.Main(String[] args) in C:\projects\nswag\src\NSwag\Program.cs:line 28
1>  ---> (Inner Exception #0) System.InvalidOperationException: Could not find item type of array type 'System.Web.Mvc.TempDataDictionary'.
1>  
1>  Server stack trace: 
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.LoadProperty(Type parentType, PropertyInfo property, JsonSchema4 parentSchema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GeneratePropertiesAndInheritance(Type type, JsonSchema4 schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.RootTypeJsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\RootTypeJsonSchemaGenerator.cs:line 45
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.RootTypeJsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\RootTypeJsonSchemaGenerator.cs:line 52
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.LoadProperty(Type parentType, PropertyInfo property, JsonSchema4 parentSchema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GeneratePropertiesAndInheritance(Type type, JsonSchema4 schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NJsonSchema.Generation.JsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.RootTypeJsonSchemaGenerator.GenerateObject[TSchemaType](Type type, TSchemaType schema, JsonSchema4 rootSchema, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\RootTypeJsonSchemaGenerator.cs:line 45
1>     at NJsonSchema.Generation.JsonSchemaGenerator.Generate[TSchemaType](Type type, JsonSchema4 rootSchema, IEnumerable`1 parentAttributes, ISchemaDefinitionAppender schemaDefinitionAppender, ISchemaResolver schemaResolver)
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.CreateAndAddSchema(SwaggerService service, Type type, IEnumerable`1 parentAttributes, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 499
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.LoadReturnType(SwaggerService service, SwaggerOperation operation, MethodInfo method, ISchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 452
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.GenerateForController(SwaggerService service, Type controllerType, String excludedMethodName, SchemaResolver schemaResolver) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 98
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiToSwaggerGenerator.GenerateForControllers(IEnumerable`1 controllerTypes, String excludedMethodName) in C:\projects\nswag\src\NSwag.Core\CodeGeneration\SwaggerGenerators\WebApi\WebApiToSwaggerGenerator.cs:line 72
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiAssemblyToSwaggerGenerator.WebApiAssemblyLoader.GenerateForControllers(IEnumerable`1 controllerClassNames, String settingsData) in C:\projects\nswag\src\NSwag.CodeGeneration\SwaggerGenerators\WebApi\WebApiAssemblyToSwaggerGenerator.cs:line 89
1>     at System.Runtime.Remoting.Messaging.StackBuilderSink._PrivateProcessMessage(IntPtr md, Object[] args, Object server, Object[]& outArgs)
1>     at System.Runtime.Remoting.Messaging.StackBuilderSink.SyncProcessMessage(IMessage msg)
1>  
1>  Exception rethrown at [0]: 
1>     at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
1>     at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiAssemblyToSwaggerGenerator.WebApiAssemblyLoader.GenerateForControllers(IEnumerable`1 controllerClassNames, String settingsData)
1>     at NSwag.CodeGeneration.SwaggerGenerators.WebApi.WebApiAssemblyToSwaggerGenerator.GenerateForControllers(IEnumerable`1 controllerClassNames) in C:\projects\nswag\src\NSwag.CodeGeneration\SwaggerGenerators\WebApi\WebApiAssemblyToSwaggerGenerator.cs:line 62
1>     at NSwag.Commands.WebApiToSwaggerCommand.<RunAsync>d__32.MoveNext() in C:\projects\nswag\src\NSwag\Commands\WebApiToSwaggerCommand.cs:line 97
1>  --- End of stack trace from previous location where exception was thrown ---
1>     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
1>     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
1>     at NSwag.Commands.WebApiToSwaggerCommand.<RunAsync>d__31.MoveNext() in C:\projects\nswag\src\NSwag\Commands\WebApiToSwaggerCommand.cs:line 84
1>  --- End of stack trace from previous location where exception was thrown ---
1>     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
1>     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
1>     at NConsole.CommandLineProcessor.<ProcessAsync>d__8.MoveNext()<---
```

**Comment by rsuter** [comment]
The current logic selects all classes which end with "Controller" or inherit from ApiController, is it possible that something wrong is selected?

**Comment by rsuter** [comment]
FYI: All controllers which are listed in the NSwagStudio are selected...

**Comment by CastleSoft** [comment]
It's the default stub generated using VS2015Up2.   It shows all 3 controllers in the GUI, and if I name each one on the command line it works. But when I try to use the command line leaving /controller or /controllers empty. It fails with the above message.

**Comment by rsuter** [comment]
Do you have the type TempDataDictionary somewhere in the public API of these controllers?

**Comment by rsuter** [comment]
Can you try again with v2.23?

**Comment by rsuter** [comment]
If this was a .NET Core project, it should be fixed in v2.27

**Comment by rsuter** [comment]
Cmd line and controller class selection has been fixed in v2.33
